docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md

```diff
@@ -0,0 +1,158 @@
++++
+title = "Add authentication for data source plugins"
+type = "docs"
+aliases = ["/docs/grafana/latest/plugins/developing/auth-for-datasources/",  "/docs/grafana/latest/developers/plugins/authentication/"]
++++
+
+# Add authentication for data source plugins
+
+This page explains how to authenticate against an third-party API from a data source plugin using the Grafana data source proxy.
+
+When a user saves a password, or any other sensitive data, as a data source option, Grafana encrypts the data and stores it in the Grafana database. Any encrypted data source options can only be decrypted on the Grafana server. This means that any data source that makes authenticated queries needs to request the decrypted data to be sent to the browser.
+
+To minimize the amount of sensitive information sent to and from the browser, data source plugins can use the Grafana _data source proxy_. When using the data source proxy, any requests containing sensitive information go through the Grafana server. No sensitive data will be sent to the browser after the data has been saved.
+
+Some data sources, like [Prometheus](https://grafana.com/docs/grafana/latest/features/datasources/prometheus/) and [InfluxDB](https://grafana.com/docs/grafana/latest/features/datasources/influxdb/), allow users to configure whether to use the data source proxy, through a setting called _access modes_.
+
+## Add a proxy route to your plugin
+
+To forward requests through the Grafana proxy, you need to configure one or more _routes_. A route specifies how the proxy will transform outgoing requests. All routes for a given plugin are defined in the [plugin.json]({{< relref "metadata.md" >}}) file.
+
+Let's add a route to proxy requests to `https://api.example.com/foo/bar`.
+
+1. Add the route to `plugin.json`.
+
+   ```json
+   "routes": [
+     {
+       "path": "example",
+       "url": "https://api.example.com"
+     }
+   ]
+   ```
+
+1. In the `DataSource`, extract the proxy URL from `instanceSettings` to a class property called `url`.
+
+   ```ts
+   export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
+     url?: string;
+
+     constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>) {
+       super(instanceSettings);
+
+       this.url = instanceSettings.url;
+     }
+
+     // ...
+
+   }
+   ```
+
+1. In the `query` method, make a request using [BackendSrv]({{< relref "../../packages_api/runtime/backendsrv.md" >}}).
+
+   ```ts
+   const routePath = '/example';
+
+   getBackendSrv()
+     .datasourceRequest({
+       url: this.url + routePath + '/foo/bar',
+       method: 'GET',
+     });
+   ```
+
+> You need to reload the Grafana server every time you make a change to your `plugin.json` file.
```

Comment:
Is this a note? If so, it should start with "Note:" If not, then why is it indented?

Also, this information seems like it should be before you make changes to plugin.json rather than after.